cli/auth.go

```diff
@@ -320,6 +320,180 @@ func (c *viamClient) createOrganizationAPIKey(orgID, keyName string) (*apppb.Cre
 	return c.client.CreateKey(c.c.Context, req)
 }
 
+// LocationAPIKeyCreateAction corresponds to `location api-key create`.
+func (c *viamClient) LocationAPIKeyCreateAction(cCtx *cli.Context) error {
+	c, err := newViamClient(cCtx)
+	if err != nil {
+		return err
+	}
+
+	err = c.locationAPIKeyCreateAction(cCtx)
+	return err
+}
+
+func (c *viamClient) locationAPIKeyCreateAction(cCtx *cli.Context) error {
+	if err := c.ensureLoggedIn(); err != nil {
+		return err
+	}
+
+	locationID := cCtx.String(dataFlagLocationID)
+	orgID := cCtx.String(dataFlagOrgID)
+	keyName := cCtx.String(apiKeyCreateFlagName)
+
+	if locationID == "" {
+		return errors.New("cannot create an api-key for a location without an ID")
+	}
+
+	if keyName == "" {
+		// Default name is in the form myusername@gmail.com-2009-11-10T23:00:00Z
+		// or key-uuid-2009-11-10T23:00:00Z if it was created by a key
+		keyName = fmt.Sprintf("%s-%s", c.conf.Auth, time.Now().Format(time.RFC3339))
+		infof(cCtx.App.Writer, "using default key name of %s", keyName)
+	}
+
+	if orgID == "" {
```

Comment:
This where I added some logic that could be argued against. To streamline the UX of this, I thought it could be cool if we could try and "guess" the OrgID from the location if its not provided and there is only a single org on the location i.e. not a share location.

Reply:
That seems like fair logic to me.

Reply:
oo i was picturing this happening on the server side actually. i would also argue that the default naming should happen server side (although i know that already exists for org api keys). especially given that something like this that theoretically should behave the same across all clients (ex: i dont see a reason why we wouldn't want to add this checking for calls from the client or why the default name for keys created via the client should be different). food for thought though as its not a huge deal

Reply:
doing all this server side is slightly more performant as well since it limits round trip requests

Reply:
Yeah ^ agree with the performant stuff, it was an easier change to do it in here but I agree. Let me look at putting in the PR where we need to better check perms for APIKeyCreate and see if that makes sense.

Reply:
>  doing all this server side is slightly more performant as well since it limits round trip requests

Ehh, seems like it would still be pretty fast and API key creation does not seem like it would require super low latency...

If it is easier / cleaner to do on the backend then it makes sense, but a few more gRPC calls seems cheap compared to a few hours of dev time.

Reply:
Time to list my robots: 0.439s
```sh
[zack@rama:~/work/rdk/cli]$ time ./main robot list
Dan's Org -> Rover Rental
snowy-morning (id: 7af3e398-249a-45ff-a785-bb3e37c78ff6)
young-violet (id: a21a53e2-2e11-4f64-b47f-e29967279f94)

real	0m0.439s
user	0m0.054s
sys	0m0.030s
```
Time to create a robot api key: 0.545 s
```sh
time ./main robot api-key create --robot-id 7af3e398-249a-45ff-a785-bb3e37c78ff6
real	0m0.545s
user	0m0.036s
sys	0m0.021s
```

The performance impact of these calls is minute

Reply:
thanks Zack! yeah based on that ^ it's not a huge issue. May just be nicer to put this in APP though if we want to support doing stuff like this across multiple resources.

Reply:
yeah i totally dont disagree that in absolute numbers (and in ideal conditions) this extra request is no big deal. if im understanding the architecture correctly though (and please correct me if i'm not, still ramping!), there could be a user in australia using the cli and each of these network request would have to hit our app servers in us-east-1 (or whatever the GCP equivalent is). in that case, an extra round trip call could make a slightly bigger difference. 

again, don't think its a huge deal in practice right now but wanted to bring it up from a code hygiene/best practices perspective to try to limit these api calls where appropriate as it may be more applicable for other stuff later 🤷‍♀️ 

feel free to burn after reading

Reply:
Made an APP PR to move this guessing logic over there as had to make a change there anyways. This may be useful down the line for other resources too so made sense to just move in there for cleaner code.